ludwig/schema/features/utils.py

```diff
@@ -42,44 +53,36 @@ def get_input_feature_jsonschema(model_type: str):
 
     Returns: JSON Schema
     """
-    input_feature_types = sorted(list(input_config_registry.keys()))
+    input_feature_types = sorted(list(input_config_registry(model_type).keys()))
     schema = {
-        "type": "array",
-        "minItems": 1,
-        "items": {
-            "type": "object",
-            "properties": {
-                "name": {"type": "string", "title": "name", "description": "Name of the input feature."},
-                "type": {
-                    "type": "string",
-                    "enum": input_feature_types,
-                    "title": "type",
-                    "description": "Type of the input feature",
-                },
-                "column": {"type": "string", "title": "column", "description": "Name of the column."},
+        "type": "object",
```

Comment:
Good question, it should get it from the `FeaturesTypeSelection` which has `min_length=1` set by defaults. But let me check.

Reply:
Yes, looks like it works. Added tests to verify.

Reply:
Awesome that it works, also verified it locally. I'm a little confused about how this works though... the custom schema here is no longer even an array so how does it end up being one in the final assembly (let alone how does the `allOf` end up in the right place)?

Reply:
I think it's `marshmallow-jsonschema` magic.

Reply:
@tgaddair I'm not so sure. Whatever is set in `_json_type_mapping` should override any kind of automatic logic so at minimum while I might expect it to perhaps fix the `type` to be an `array` I would not expect it to insert `items` or know to cherry-pick `allOf` to one level higher than it is in the final schema. But I actually wanted to talk about this in our next sync

Reply:
I just figured this out. Despite the fact that `json_type_mapping` is _supposed_ to be the ultimate override method, `marshmallow_jsonschema` _will_ change the type of the field if the class in question derives a native marshmallow type. Since `FeaturesTypeSelection` ultimately derives `marshmallow.fields.List`, the package _will_ force it to be a list - in particular by taking the output of `json_type_mapping` and shoving it inside of a standard list's JSON schema (that is where the surrounding `type: array : { items: { ...` comes from).

We are lucky that all of the feature's schema attributes and conditional logic indeed are supposed to be nested inside of `items`, otherwise this would have caused a major error!

Reply:
Very interesting! Nice discovery.